huaweicloud/data_source_huaweicloud_identity_custom_role.go

```diff
@@ -0,0 +1,112 @@
+package huaweicloud
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/huaweicloud/golangsdk/openstack/identity/v3.0/policies"
+)
+
+func DataSourceIdentityCustomRole() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceIdentityCustomRoleRead,
+
+		Schema: map[string]*schema.Schema{
+			"domain_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"references": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
```

Comment:
`id` is the system attribute which  do not define manually.

Reply:
It's better if id can be used as a query condition. So I turned the id into an optional parameter.